src/ast.ts

```diff
@@ -174,6 +174,30 @@ export abstract class Node {
     return new TypeParameterNode(name, extendsType, defaultType, range);
   }
 
+  static createCompoundNamedType(
+    identifiers: IdentifierExpression[],
+    typeParameters: TypeNode[] | null,
+    isNullable: bool,
+    range: Range
+  ): NamedTypeNode {
+    let length = identifiers.length;
```

Comment:
nit: Top-level vars are `var` in compiler code, mostly historically and to aid the debugging the bootstrap, whereas `let` indicates a shared local in an inner block.